swift/apple/FirezoneKit/Sources/FirezoneKit/Views/MenuBar.swift

```diff
@@ -446,4 +455,14 @@
       isMenuVisible = false
     }
   }
+
+  extension MenuBar: AuthStoreAlertDelegate {
```

Comment:
Our end-users will be very non-technical. I worry showing them non-actionable dialogs like this and having another state the app can be in (signed in but disconnected due to unknown reasons) just creates confusion.

Would it make sense to do the following instead?

- For OS-initiated disconnects, attempt to reconnect automatically N (say, 3) times (sleeping a couple hundred milliseconds in between), printing an error to the log each time
- If that fails, print a final error log, and just sign them out

If the tunnel can't be reconnected automatically, I'm not sure it makes sense to keep the app in the signed in state.

We have to have a consistent UX across all apps so we would also need this experience for Windows/Linux/Android/etc so I'm really trying to keep it as simple as possible for now.

Reply:
I realize the above scenario creates an edge case where the user can't toggle the VPN off in System Settings because it will automatically reconnect.

I think that edge case is fine for a first release though. Other products behave similarly.

The other option for this is to try and be smarter about which `NEProviderStopReason`s we reconnect automatically on. It would be very annoying for example to the user to have to reconnect the tunnel manually after each `.sleep`

Reply:
As discussed in the call, implemented this:
- If the user disconnects (NEProviderStopReason == .userInitiated), we sign out the user
- If the OS disconnects us for some reason, we try reconnecting 3 times, and if it fails the third time, we sign out the user